prql-compiler/src/semantic/context.rs

```diff
@@ -163,20 +163,10 @@ impl Context {
                 Ok(module_ident + Ident::from_name(ident.name.clone()))
             }
 
-            // don't report ambiguous variable, database may be able to resolve them
+            // ambiguous
             _ => {
-                // insert default
-                let ident = NS_NO_RESOLVE.to_string();
-                self.root_mod
-                    .names
-                    .insert(ident, Decl::from(DeclKind::NoResolve));
-
-                log::debug!(
-                    "... could either of {:?}",
-                    decls.iter().map(|x| x.to_string()).collect_vec()
-                );
-
-                Ok(Ident::from_name(NS_NO_RESOLVE))
+                let decls = decls.into_iter().map(|d| d.to_string()).join(", ");
+                return Err(format!("Ambiguous reference. Could be from any of {decls}"));
```

Comment:
Do you have a test case for this error? I'd quite like to see an example of the message that it produces.

Reply:
No I don't, maybe I should I one.

```
from x
join y []
select a
Error: 
   ╭─[_a.prql:3:8]
   │
 3 │ select a
   ·        ┬  
   ·        ╰── Ambiguous name. Could be from any of _frame.y.`*`, _frame.x.`*`
───╯
```

Reply:
FYI I started error message tests; need to fix the whitespace issue and then it should be easy to build off: https://github.com/prql/prql/pull/1181